### PR TITLE
FEATURE: Upgrade analytics.js to gtag

### PR DESCRIPTION
**Pull request by justindirose**
This is a reimplementation of https://github.com/discourse/discourse/pull/10893 with a few fixes in mind. This will be a breaking change for sites who use any custom events in theme components.

Per [Google](https://developers.google.com/analytics/devguides/collection/upgrade/analyticsjs), sites are encouraged to upgrade from `analytics.js` to `gtag.js` for Google Analytics tracking. This does leave in the page tracking call for using `analytics.js` and `window.ga. If someone wanted to stay on the old tag manually via a theme component, the page tracking would work correctly still.

### CODE REVIEW

**Comment by davidtaylorhq** [comment]
(We discussed this elsewhere, but posting for reference)

It would be cool if this change was behind a site setting to start with, so we can roll it out slowly and test it along the way.

**Comment by justindirose** [comment]
@davidtaylorhq did some cleanup here in addition to adding the discussed site setting. Tested locally and it appears the script works for both v3 and v4 (could see realtime user logged in and page tracking). Added some logic to avoid double tracking pageviews if we're on v4 since `window.ga` is still accessible.

**Comment by davidtaylorhq** [comment]
Looking good to me! Last thing to check is that it works when assets are compiled in production mode. I think we will need an update here for the renamed v3, and to add the v4:

https://github.com/discourse/discourse/blob/0863c36221130b161d63562a58ac97171475dca5/config/application.rb#L162-L162

and here:

https://github.com/discourse/discourse/blob/0863c36221130b161d63562a58ac97171475dca5/lib/discourse_js_processor.rb#L55-L55

(I did a search for `google-universal-analytics` across the codebase to find those)

**Comment by justindirose** [comment]
@davidtaylorhq should be fixed up. Did another search myself and didn't come across anything else.

File: config/site_settings.yml [hunk 1]
```diff
@@ -139,6 +139,13 @@ basic:
     default: 365
     min: 7
     max: 36500
+  ga_version:
+    type: enum
+    default: v3_analytics
+    choices:
+      - v3_analytics
+      - v4_gtag
+    client: true
```
Comment (davidtaylorhq) [inline config/site_settings.yml:148]: Is this actually used anywhere in the client javascript? As far as I can see it's just used in the ruby/erb code, so we should be able to remove the `client: true` line.

**Review by davidtaylorhq — APPROVED** [review]

**Pull request merged**